components/brave_vpn/common/wireguard/wireguard_utils.cc

```diff
@@ -82,6 +87,77 @@ WireguardKeyPair GenerateNewX25519Keypair() {
       EncodeBase64(std::vector<uint8_t>(privkey, privkey + 32)));
 }
 
+bool ValidateKey(const wchar_t* key,
+                 std::string* output,
+                 const std::string field_name) {
+  if (!base::WideToUTF8(key, wcslen(key), output)) {
+    VLOG(1) << "failed decoding " << field_name;
+    return false;
+  }
+
+  base::TrimWhitespaceASCII(*output, base::TRIM_ALL, output);
+
+  if ((*output).length() == 0) {
+    VLOG(1) << field_name << " does not have a value";
+    return false;
+  }
+
+  return true;
+}
+
+bool ValidateAddress(const wchar_t* address, std::string* output) {
+  if (!base::WideToUTF8(address, wcslen(address), output)) {
+    VLOG(1) << "failed decoding address";
+    return false;
+  }
+
+  base::TrimWhitespaceASCII(*output, base::TRIM_ALL, output);
+
+  auto parsed = net::IPAddress::FromIPLiteral(*output);
+  if (!parsed.has_value()) {
+    VLOG(1) << "failed parsing address";
+    return false;
+  }
+
+  auto parsed_ip = parsed.value();
+  if (!parsed_ip.IsValid()) {
+    VLOG(1) << "address is not valid";
+    return false;
+  }
+
+  if (!parsed_ip.IsIPv4()) {
+    VLOG(1) << "address must be IPv4";
+    return false;
+  }
+
+  return true;
+}
+
+bool ValidateEndpoint(const wchar_t* endpoint, std::string* output) {
+  if (!base::WideToUTF8(endpoint, wcslen(endpoint), output)) {
+    VLOG(1) << "failed decoding endpoint";
+    return false;
+  }
+
+  base::TrimWhitespaceASCII(*output, base::TRIM_ALL, output);
+
+  std::string parsed_host;
+  int parsed_port = 0;
+  if (!net::ParseHostAndPort(*output, &parsed_host, &parsed_port)) {
+    VLOG(1) << "failed parsing endpoint";
+    return false;
+  }
+
+  GURL parsed_host_url = GURL("https://" + parsed_host);
+  if (!parsed_host_url.DomainIs("guardianapp.com") &&
+      !parsed_host_url.DomainIs("sudosecuritygroup.com")) {
```

Comment:
Here's the hostname validate @diracdeltas

Reply:
@bsclifton i think you can just check the host using [DomainIs](https://source.chromium.org/chromium/chromium/src/+/main:url/url_util.h;bpv=1;bpt=1;l=208?gsn=DomainIs&gs=KYTHE%3A%2F%2FKp4CCowBa3l0aGU6Ly9jaHJvbWl1bS5nb29nbGVzb3VyY2UuY29tL2NvZGVzZWFyY2gvY2hyb21pdW0vc3JjLy9tYWluP2xhbmc9YyUyQiUyQj9wYXRoPXVybC91cmxfdXRpbC5oI040enVpVEFHblg1Q3lJbkYzMmhfaUVqLTgwZlkzWGNZcEtFRV9BaHBsbmMKjAFreXRoZTovL2Nocm9taXVtLmdvb2dsZXNvdXJjZS5jb20vY29kZXNlYXJjaC9jaHJvbWl1bS9zcmMvL21haW4_bGFuZz1jJTJCJTJCP3BhdGg9dXJsL3VybF91dGlsLmgjV0JnU043SnVsR0d6bjZUYllEMkRraEZBbEZac1E1bjgzb0tKQjU1aFhrVQ%3D%3D) and not have to prepend the https? but fine either way

Reply:
Good call! Fixed in https://github.com/brave/brave-core/pull/24261/commits/dea24007e9c5b463ab986f663106c63975a4c76a